### ISSUE TITLE
Typescript definitions are not tested

### ISSUE THREAD

**Issue by rosskevin** [i18next/react-i18next#654]
**Describe the bug**
Downstream in my app, since the types were brought in-repo - I now have typescript problems related to i18next/react-i18next.  This may be a local problem, or one with the repo, but without tests in the repo I have no certainty.

Tests were used in the @types/react-i18next but never brought here e.g. [`react-i18next-tests.tsx`](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29945/files)

**Occurs in react-i18next version**
 react-i18next  ^9.0.0
 i18next   ^13.1.0

**TL;DR**

I'm working on a PR with typescript usage tests that will yield some certainty that types will work as represented by examples.  With that in place, it will give me confidence and I can focus on potential userland errors in my environment.

**Comment by Tsury** [comment]
Same, getting TS issues:

```
node_modules/react-i18next/index.d.ts:3:41 - error TS2307: Cannot find module 'create-react-context'.
import { Context as ReactContext } from 'create-react-context';
```

**Comment by rosskevin** [comment]
PR is now good to merge.

**Comment by jamuhl** [comment]
published in react-i18next@9.0.1
